snapcraft/plugins/python.py

```diff
@@ -0,0 +1,235 @@
+
+"""The python plugin can be used for python 2 or 3 based parts.
+
+The python plugin can be used for python projects where you would
+want to do:
+
+    - import python modules with a requirements.txt
+    - build a python project that has a setup.py
+    - install packages straight from pip
+
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+
+Additionally, this plugin uses the following plugin-specific keywords:
+
+    - requirements:
+      (string)
+      Path to a requirements.txt file
+    - constraints:
+      (string)
+      Path to a constraints file
+    - process-dependency-links:
+      (bool; default: false)
+      Enable the processing of dependency links.
+    - python-packages:
+      (list)
+      A list of dependencies to get from PyPi
+    - python-version:
+      (string; default: python3)
+      The python version to use. Valid options are: python2 and python3
+"""
+
+import os
+import re
+import subprocess
+from glob import glob
+
+import snapcraft
+from snapcraft import common
+
+
+class PythonPlugin(snapcraft.BasePlugin):
+
+    @classmethod
+    def schema(cls):
+        schema = super().schema()
+        schema['properties']['requirements'] = {
+            'type': 'string',
+        }
+        schema['properties']['constraints'] = {
+            'type': 'string',
+        }
+        schema['properties']['python-packages'] = {
+            'type': 'array',
+            'minitems': 1,
+            'uniqueItems': True,
+            'items': {
+                'type': 'string'
+            },
+            'default': [],
+        }
+        schema['properties']['process-dependency-links'] = {
+            'type': 'boolean',
+            'default': False,
+        }
+        schema['properties']['python-version'] = {
+            'type': 'string',
+            'default': 'python3',
+        }
+        schema.pop('required')
+
+        # Inform Snapcraft of the properties associated with pulling. If these
+        # change in the YAML Snapcraft will consider the pull step dirty.
+        schema['pull-properties'].extend([
+            'requirements',
+            'constraints',
+            'python-packages',
+            'python-version',
+        ])
+
+        return schema
+
+    @property
+    def plugin_build_packages(self):
+        if self.options.python_version == 'python3':
+            return [
+                'python3-dev',
+                'python3-pip',
+                'python3-pkg-resources',
+                'python3-setuptools',
+            ]
+        elif self.options.python_version == 'python2':
+            return [
+                'python-dev',
+                'python-pip',
+                'python-pkg-resources',
+                'python-setuptools',
+            ]
+        else:
+            return []
```

Comment:
raise an exception? This should never happen.
Now this is a good use for assert :p